### PR TITLE
qemu.tests: Install the kernel dependencies packages

### PR DESCRIPTION
**Pull request by spcui**
For installing the latest rhel kernel package, failed with
dracut dependencies, so add "--nodeps" to skip it.

Signed-off-by: Shuping Cui <scui@redhat.com>

### CODE REVIEW

**Comment by spcui** [comment]
ID: 1245007

**Comment by spcui** [comment]
# ./tools/scan_results.py 
Test                                                                                                                                            Status     Seconds  Info
----                                                                                                                                            ------     -------  ----
        (Result file: /root/staf-kvm-devel/autotest-devel/client/results/default/status)
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.1.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.rh_kernel_update    GOOD       320      completed successfully
----                                                                                                                                            GOOD       342

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang <shuang@redhat.com>

**Comment by CongSmile** [comment]
Not sure this is a good way to solve this problem.

**Comment by vipmike007** [comment]
Agree w/ @CongSmile , better to slove the kernel dependencies instead of force install it.

**Comment by xutian** [comment]
@CongSmile  and @vipmike007 

Any suggest to solve dependencies  nicely ?

**Comment by CongSmile** [comment]
I think the root cause it the tree / package problems, we can't just use "nodeps" to solve the problem.
If we force to install the package, the guest may can't boot up.

**Comment by vipmike007** [comment]
@xutian Try yum install kernelxxxx.rpm -y ,it should resolve the dependencies.if the repo is correct in the guest.

**Comment by spcui** [comment]
Hi @CongSmile @xutian @vipmike007 updated now, could you help review again?
Thanks

**Comment by spcui** [comment]
# ../../tools/scan_results.py 
Test                                                                                                                                            Status     Seconds  Info
----                                                                                                                                            ------     -------  ----
        (Result file: /root/staf-kvm-devel/autotest-devel/client/results/default/status)
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.6.5.i386.io-github-autotest-qemu.rh_kernel_update      GOOD       362      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.6.5.x86_64.io-github-autotest-qemu.rh_kernel_update    GOOD       381      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.6.6.i386.io-github-autotest-qemu.rh_kernel_update      GOOD       425      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.6.6.x86_64.io-github-autotest-qemu.rh_kernel_update    GOOD       419      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.rh_kernel_update    GOOD       339      completed successfully

File: qemu/tests/rh_kernel_update.py [hunk 1]
```diff
@@ -330,6 +329,13 @@ def upgrade_guest_pkgs(session, pkg, debuginfo=False,
             if params.get("install_rpm_from_local") == "yes":
                 rpm_install_func = copy_and_install_rpm
 
+            kernel_deps_pkgs = params.get("kernel_deps_pkgs").split()
+            if kernel_deps_pkgs:
+                for pkg in kernel_deps_pkgs:
+                    arch = params.get("arch_dracut",
```
Comment (xutian) [inline qemu/tests/rh_kernel_update.py]: "arch_%s" % pkg ?

Comment (spcui) [inline qemu/tests/rh_kernel_update.py]: sorry for mistake, update now, thanks

Comment (xutian) [inline qemu/tests/rh_kernel_update.py]: We can upgrade dracut like upgrade glibc

Comment (spcui) [inline qemu/tests/rh_kernel_update.py]: Yes, but for dracut on RHEL.6 guest, need use '--arch=noarch' get brew link, like this 'brew latest-pkg xxx dracut --arch=noarch --paths', so still need arch_dracut param.

Comment (xutian) [inline qemu/tests/rh_kernel_update.py]: On 08/04/2015 02:42 PM, Shuping Cui wrote:
>
> In qemu/tests/rh_kernel_update.py 
> <https://github.com/autotest/tp-qemu/pull/419#discussion_r36161174>:
>
> > @@ -330,6 +329,13 @@ def upgrade_guest_pkgs(session, pkg, debuginfo=False,
> >              if params.get("install_rpm_from_local") == "yes":
> >                  rpm_install_func = copy_and_install_rpm
> >
> > +            kernel_deps_pkgs = params.get("kernel_deps_pkgs").split()
> > +            if kernel_deps_pkgs:
> > +                for pkg in kernel_deps_pkgs:
> > +                    arch = params.get("arch_dracut",
>
> Yes, but for dracut on RHEL.6 guest, need use '--arch=noarch' get brew 
> link, like this 'brew latest-pkg xxx dracut --arch=noarch --paths', so 
> still need arch_dracut param.
>
We can set all depend packages use param like "upgrade_pkgs", and 
pkg_arch should be configurable not limited to arch_dracut.

Comment (xutian) [inline qemu/tests/rh_kernel_update.py]: We can use params.object_params(pkg).get(arch) to get arch param for each packages, not limited your code to dracut package only.

**Comment by xutian** [comment]
Merged, thanks @spcui @CongSmile  and @suqinhuang

**Pull request merged**